theseus/core/objective.py

```diff
@@ -503,3 +503,20 @@ def to(self, *args, **kwargs):
         self.dtype = dtype or self.dtype
         if self._vectorization_to is not None:
             self._vectorization_to(*args, **kwargs)
+
+    def step_optim_vars(
```

Comment:
Nice, Objective class actually seems like a good place for this function.

`step_...` makes it sound like optimizer related, how about `update_...`? Maybe `update_optim_vars_with_delta` if we want to be really descriptive.

Reply:
Maybe `retract_optim_vars`?

Reply:
Yeah `retract_optim_vars` is good.

Reply:
Changed.